src/Common/src/System/StrongToWeakReference.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    /// <summary>Provides an object wrapper that can transition between strong and weak references to the object.</summary>
+    internal sealed class StrongToWeakReference<T> : WeakReference where T : class
```

Comment:
Yeah, I don't know.  I started with ```WeakReference<T>``` and then backed away from it primarily because it was sealed.  Secondarily, its access patterns would have led to slightly more convoluted code, as it forces you to use a TryGetTarget pattern when Target really is what I wanted.